dspace-api/src/main/java/org/dspace/identifier/FilteredIdentifierProvider.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.identifier;
+
+import java.sql.SQLException;
+
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.logic.Filter;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This abstract class adds extra method signatures so that implementing IdentifierProviders can
+ * handle "skip filter" booleans, so that any configured filters can be skipped and DOI registration forced.
+ *
+ * @author Kim Shepherd
+ * @version $Revision$
+ */
+public abstract class FilteredIdentifierProvider extends IdentifierProvider {
+
+    protected Filter filterService;
+
+    /**
+     * Setter for spring to set the filter service from the property in configuration XML
+     * @param filterService - an object implementing the org.dspace.content.logic.Filter interface
+     */
+    @Autowired
+    public void setFilterService(Filter filterService) {
+        this.filterService = filterService;
+    }
+
+    /**
+     * Register a new identifier for a given DSpaceObject
+     * @param context    - DSpace context
+     * @param dso        - DSpaceObject to use for identifier registration
+     * @param skipFilter - boolean indicating whether to skip any filtering of items before performing registration
+     * @return identifier
+     * @throws IdentifierException
+     */
+    public abstract String register(Context context, DSpaceObject dso, Boolean skipFilter)
+        throws IdentifierException;
+
+    /**
+     * Register a specified identifier for a given DSpaceObject
+     * @param context    - DSpace context
+     * @param dso        - DSpaceObject identified by the new identifier
+     * @param identifier - String containing the identifier to register
+     * @param skipFilter - boolean indicating whether to skip any filtering of items before performing registration
+     * @throws IdentifierException
+     */
+    public abstract void register(Context context, DSpaceObject dso, String identifier, Boolean skipFilter)
```

Comment:
better to use a primitive boolean

Reply:
see my other comment near the canMint section - a null or missing filter should not default to false, so I didn't want to use a primitive here

perhaps this also means i have confused things iwth my naming... would it be clearer code to read if there wa a primitive boolean called applyFilter instead of skipFilter? then it would be save since a null or missing filter would mean the default behaviour was applyFilter == false which is good.
currently, if skipFilter is a primitive and ends up false for some non-logical reason (null or missing filter, some other error causes null result to be returend) then the filter is still applied and that would definitely cause problems.